examples/enc-search-sample.rs

```diff
@@ -0,0 +1,421 @@
+// =============================================================================
+// This is a sample program in Rust that uses the ironoxide SDK to illustrate
+// various patterns for using the encrypted search features.
+//
+// To execute this program, you need to provide a file that contains a device
+// context. This file can be generated using the ironoxide-cli utility. You can
+// download this from https://github.com/IronCoreLabs/ironoxide-cli/releases;
+// if a compiled version for your platform or architecture is not available, you
+// can use cargo to build and install it. The program assumes the device
+// context file is named "example.devcon" in the current directory.
+//
+// Also note that this program creates two different groups - in order to avoid
+// errors when executing the program multiple times, it appends the current
+// time (as a time_t) when the program is started to the group IDs.
+//
+// Copyright (c) 2020  IronCore Labs, Inc.
+// =============================================================================
+
+use ironoxide::prelude::*;
+use lazy_static::lazy_static;
+use mut_static::MutStatic;
+use std::{
+    convert::TryFrom,
+    fmt,
+    fs::File,
+    path::PathBuf,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+type Result<T> = std::result::Result<T, AppErr>;
```

Comment:
You might consider looking at https://github.com/dtolnay/anyhow to reduce the error handling code in this sample. I think it would allow you to get rid of all the `From` instances at the bottom of the file.

Reply:
Maybe in a future PR. I borrowed this from `ironoxide-cli`.

Reply:
Done